modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormFieldSettings.es.js

```diff
@@ -70,14 +70,17 @@ const StateSync = ({
 	}, [dispatch, activePage]);
 
 	useEffect(() => {
-		dispatch({
-			payload: {
-				activePage: focusedField?.settingsContext.currentPage,
-				field: focusedField,
-			},
-			type: CORE_EVENT_TYPES.FIELD.CLICK,
-		});
-	}, [dispatch, focusedField]);
+		if (initialFocusedField.instanceId !== focusedField.instanceId) {
+			setFocusedField(initialFocusedField);
+			dispatch({
+				payload: {
+					activePage: initialFocusedField.settingsContext.currentPage,
+					field: initialFocusedField,
+				},
+				type: CORE_EVENT_TYPES.FIELD.CLICK,
+			});
+		}
+	}, [dispatch, initialFocusedField, focusedField.instanceId]);
```

Comment:
You did a great work by cut down the amount of unnecessary dispatches made! 

But I'm wondering if we really need this `useEffect()`, cause the dispatch of the `EVENT_TYPES.FIELD.CLICK` is probably being triggered somewhere else. I don't see the `FormFieldSettings.es.js` as the component responsible for make it happen.

Can you check if things still working if you remove that? 

Actually, there are a lot of other dispatches happening  here that I'm not sure if they are properly placed. But anyway you did a great improvement and since this code are much better than the previous one  I'm fine to approve that.

Reply:
Hi @miltonmc! Of course, I will see this and come back here.
Thanks for your comments.

Reply:
I commented this code and deployed the module. I didn't find problems after this alteration. I ran the test suite this module and all passed. What do you think?

Reply:
So, I think we can get rid of this `useEffect()`